app/Console/Commands/ImportBusinessesSpreadsheet.php

```diff
@@ -0,0 +1,219 @@
+        $conn = $em->getConnection();
+        $swlat = 51.32990102458417;
+        $swlng = -5.371793182658548;
+        $nelat = 53.46775372527346;
+        $nelng = -2.6032384951585485;
+
+        $stmt = $conn->prepare("DELETE FROM businesses WHERE X(geolocation) >= $swlng AND X(geolocation) <= $nelng AND Y(geolocation) >= $swlat AND Y(geolocation) <= $nelat;");
+        $stmt->execute();
+
+        // load the CSV document from a file path
+        $file = $this->argument('file');
+        $this->output->writeln('Reading ' . $file);
+
+        $reader = new \PhpOffice\PhpSpreadsheet\Reader\Xlsx();
+        $spreadsheet = $reader->load($file);
+
+        // Iterate through each tab.  Last one is just info; others are one per council.
+        $councils = $spreadsheet->getSheetNames();
+        array_pop($councils);
+
+        // Ensure we don't run foul of unique index.
+        $uniques = [];
+
+        foreach ($councils as $council) {
+            $this->line("Process council $council");
+
+            $sheet = $spreadsheet->getSheetByName($council);
+
+            // First row is council name.
+            // Second row is column headings.
+            // Third and subsequent rows are data.
+            $row = 3;
+
+            $this->line("Start at " . $sheet->getCellByColumnAndRow(1, $row)->getValue());
+            while ($sheet->getCellByColumnAndRow(1, $row)->getValue()) {
+                try {
+                    // Get the fields in the first set of columns.
+                    $fields = [
+                        'name', 'address', 'postcode', 'borough', 'city', 'description', 'landline', 'mobile', 'website',
+                        'email',
+                        'ignored1', 'ignored2', 'ignored3', 'ignored4', 'ignored5',
+                        'notes', 'review_url', 'review_source', 'review_count', 'positive_percent', 'average_scores',
+                        'warranty_offered', 'warranty_details', 'publish'
+                    ];
+
+                    for ($i = 0; $i < count($fields); $i++) {
+                        $field = $fields[$i];
+                        $$field = trim($sheet->getCellByColumnAndRow($i + 1, $row)->getValue());
+                    }
+
+                    $key = $name . $address;
+
+                    if (array_key_exists($key, $uniques)) {
+                        $this->error("Duplicate business $name, $address");
+                    } else if (strlen($review_url) > 1000) {
+                        $this->error("$name has overlong review URL $review_url");
+                    } else {
+                        $uniques[$key] = TRUE;
+
+                        // Tweak the URL.
+                        if ($website) {
+                            if (stripos($website, 'http') === FALSE) {
+                                $website = "https://$website";
+                            }
+
+                            $website = preg_replace('/\/#$/', '', $website);
```

Comment:
Are these removing trailing # and /?  Could you put a comment as to what these are doing, it isn't immediately evident (perhaps my regex isn't up to scratch!)

Reply:
Will comment.